### PR TITLE
List entries

### PR DESCRIPTION
**Pull request by goodviber**
### Context
We need to allow a foster parent user to review previous diary logs for a foster child
### Changes proposed in this pull request
Index and show pages for diary entries
### Guidance to review
We need to expand our test coverage

### CODE REVIEW

File: app/assets/stylesheets/diary_entries.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/*
```
Comment (novotnyjakub) [inline app/assets/stylesheets/diary_entries.css:1]: As it's empty, I'd remove this for now.

File: app/policies/diary_entry_policy.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class DiaryEntryPolicy < ApplicationPolicy
+  def show?
+    @auth_context.role_model.is_a?(FosterParent)
+  end
+
+  class Scope < Scope
+    def resolve
+      if @auth_context.role_model.is_a?(FosterParent)
+        scope.all
```
Comment (novotnyjakub) [inline app/policies/diary_entry_policy.rb:9]: We'll need to only reduce the scope to the foster parent's placement diary entries.

```ruby
scope.joins(:placements).where(placements: { foster_parent_id: @auth_context.role_model.id }
```

Or something between these lines ☝️ 

+ as well, please add spec for this.

File: app/views/diary_entries/_foster_logs.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<%= render GovukComponent::SummaryList.new do |component| %>%
```
Comment (novotnyjakub) [inline app/views/diary_entries/_foster_logs.html.erb:1]: I personally think this will be confusing as to where a one diary entry ends and another starts if these are separate elements of the `SummaryList` component. But we can deal with it later.

File: app/views/diary_entries/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<%= render GovukComponent::SummaryList.new do |component| %>%
```
Comment (novotnyjakub) [inline app/views/diary_entries/show.html.erb:1]: Unless it was explicitly asked for, I would not put the `show` page in, until it's decided how it'll behave and what it looks like?

File: config/routes.rb [hunk 5]
```diff
@@ -12,9 +12,12 @@
 
   resources :shortlists, only: :show
   resources :placements, only: :create do
+    resources :diary_entries,
+              only: %i[index show] do
```
Comment (novotnyjakub) [inline config/routes.rb:16]: As per the comment in the template, I'd only add the `index` action here.

File: spec/requests/diary_entries_request_spec.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+require "rails_helper"
+
+RSpec.describe "DiaryEntries", type: :request do
```
Comment (novotnyjakub) [inline spec/requests/diary_entries_request_spec.rb:3]: I would remove this spec and cover the functionality in `type: :feature` one or multiple.

File: app/views/diary/steps/_select_event.html.erb [hunk 7]
```diff
@@ -6,4 +6,4 @@
   <%= f.govuk_radio_button :event, options[:emotional_health],  label: { text: t("select_event.event_options.emotional_health") }, link_errors: true %>
 <% end %>
 
-<%= f.text_field :placement_id, pattern: "[0-9]*", inputmode: "numeric", readonly: true, hidden: false, value: params[:placement_id] || f.object.placement_id%>
+<%= f.text_field :placement_id, pattern: "[0-9]*", inputmode: "numeric", readonly: true, hidden: true, value: params[:placement_id] || f.object.placement_id%>
```
Comment (novotnyjakub) [inline app/views/diary/steps/_select_event.html.erb]: If you revert this change 🙏  then this PR will not conflict with the branch I'm about to merge. I'm removing this line entirely anyway, so it's safe for you to do.

**Review by novotnyjakub — APPROVED** [review]
👍  Let's merge and fix the outstanding issues separately, so that we can demo it to the wider audience.

File: app/assets/stylesheets/diary_entries.css (hunk 1 above)
Comment (goodviber) [reply app/assets/stylesheets/diary_entries.css:1]: Done

File: app/policies/diary_entry_policy.rb (hunk 2 above)
Comment (goodviber) [reply app/policies/diary_entry_policy.rb:9]: Have had a go at it...

File: app/views/diary_entries/show.html.erb (hunk 4 above)
Comment (goodviber) [reply app/views/diary_entries/show.html.erb:1]: I think we have to take some initiative, there will most likely be a show page, as the entries are truncated on the index list, and might be a good feature for the Show and Tell

File: spec/requests/diary_entries_request_spec.rb (hunk 6 above)
Comment (goodviber) [reply spec/requests/diary_entries_request_spec.rb:3]: Ok, can take that approach
